### DISCUSSION TITLE
[错误报告]: 下载转移提示“没有获取到已完成的下载任务”

### DISCUSSION THREAD

**Discussion by vvnocode** [jxxghp/MoviePilot#409] [讨论]
### 确认

- [X] 我的版本是最新版本，我的版本号与 [version](https://github.com/jxxghp/MoviePilot/releases/latest) 相同。
- [X] 我已经 [issue](https://github.com/jxxghp/MoviePilot/issues) 中搜索过，确认我的问题没有被提出过。
- [X] 我已经 [Telegram频道](https://t.me/moviepilot_channel) 中搜索过，确认我的问题没有被提出过。
- [x] 我已经修改标题，将标题中的 描述 替换为我遇到的问题。

### 当前程序版本

1.0.2

### 问题类型

主程序运行问题

### 问题描述

下载完成后，无法自动转移。但是如果用目录监控插件可以转移。

### 发生问题时系统日志和配置文件

```bash
2023-08-19 21:43:02,639 - base.py -【ERROR】Job "同步媒体服务器 (trigger: interval[6:00:00], next run at: 2023-08-20 03:40:17 CST)" raised an exception
Traceback (most recent call last):
File "/usr/local/lib/python3.11/site-packages/apscheduler/executors/base.py", line 125, in run_job
retval = job.func(*job.args, **job.kwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/app/app/chain/mediaserver.py", line 67, in sync
for item in self.items(library.id):
File "/app/app/modules/emby/__init__.py", line 142, in mediaserver_items
tmdbid=int(item.get("tmdbid")) if item.get("tmdbid") else None,
^^^^^^^^^^^^^^^^^^^^^^^
ValueError: invalid literal for int() with base 10: '507941-section-375-marzi-ya-jabardasti'
2023-08-19 21:45:17,685 - transfer.py -【INFO】开始执行下载器文件转移 ...
2023-08-19 21:45:17,687 - transfer.py -【INFO】没有获取到已完成的下载任务
```

**Comment by vvnocode** [comment]
配置文件如下
![telegram-cloud-photo-size-1-5118679768278084722-y](https://github.com/jxxghp/MoviePilot/assets/115932197/496da291-a722-45d1-9105-df5e0bd3f5b0)

**Comment by vvnocode** [comment]
qb版本：qBittorrent v4.4.4 Web UI (64-bit)

**Comment by zraxx** [comment]
mp映射目录问题给各位朋友参考：
威联通所有硬盘都挂在/share 目录下
使用qb套件版（默认下载器）、tr容器版（保种）。1、直接映射： /share：/share。 2、qb下载目录为： /share/downloads/, 下设 /电影 和 /电视剧 2个目录； 3、媒体库（link硬链接同步），目录为/share/downloads/media，下设 /电影和/电视剧目录。所有功能正常使用，字幕插件正常。

**Comment by developer-wlj** [comment]
有没有可能，是开了自动转移做种插件，刚下载好，就赶上执行自动转移做种插件了，有可能，但概率很小。或者qb一开始就没连接上，你在首页能看到qb的状态吗

**Comment by zraxx** [comment]
自动转移做种，原文件是不变动，只是辅种重新下载种子而已。应该是qb存放目录与mp的目录不一致，导致的问题

**Comment by vvnocode** [comment]
> mp映射目录问题给各位朋友参考： 威联通所有硬盘都挂在/share 目录下 使用qb套件版（默认下载器）、tr容器版（保种）。1、直接映射： /share：/share。 2、qb下载目录为： /share/downloads/, 下设 /电影 和 /电视剧 2个目录； 3、媒体库（link硬链接同步），目录为/share/downloads/media，下设 /电影和/电视剧目录。所有功能正常使用，字幕插件正常。

下载路径和媒体路径分开可以吗？比如像我这样，

**Comment by vvnocode** [comment]
> 有没有可能，是开了自动转移做种插件，刚下载好，就赶上执行自动转移做种插件了，有可能，但概率很小。或者qb一开始就没连接上，你在首页能看到qb的状态吗

最开始没装这个插件的时候就无法转移。qb肯定是连接上了，不然没法通过moviepilot下载。

**Comment by zraxx** [comment]
建议mp和qb(主下载）、tr和jellyfin都映射目录
 /mnt: /mnt 试试
（我都映射 /share：/share）

**Comment by vvnocode** [comment]
> 建议mp和qb(主下载）、tr和jellyfin都映射目录 /mnt: /mnt 试试 （我都映射 /share：/share）

谢谢，我试试

**Comment by vvnocode** [comment]
感觉这是概率问题。有时候可以下载转移，有时候不行。插件目录同步已经被停用了的。

**Comment by developer-wlj** [comment]
> > mp映射目录问题给各位朋友参考： 威联通所有硬盘都挂在/share 目录下 使用qb套件版（默认下载器）、tr容器版（保种）。1、直接映射： /share：/share。 2、qb下载目录为： /share/downloads/, 下设 /电影 和 /电视剧 2个目录； 3、媒体库（link硬链接同步），目录为/share/downloads/media，下设 /电影和/电视剧目录。所有功能正常使用，字幕插件正常。
> 
> 下载路径和媒体路径分开可以吗？比如像我这样，

无关映射，这是qb调获取完成种子api，没获取到数据，还没走到路径映射的逻辑。qb完成的种子有打标签吗，如有标签，上传完整的log日志，有助于更好的排查

**Comment by yubanmeiqin9048** [comment]
@vvnocode 请规范tmdbid命名

**Comment by vvnocode** [comment]
> 
我开启了调试模式，这些debug级别日志是在控制台找到的。话说config目录的日志为啥还是info级别的？在哪能找到debug日志文件呢？

ERROR:   base.py - Job "同步媒体服务器 (trigger: interval[6:00:00], next run at: 2023-08-27 19:25:17 CST)" raised an exception
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/apscheduler/executors/base.py", line 125, in run_job
    retval = job.func(*job.args, **job.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/app/chain/mediaserver.py", line 67, in sync
    for item in self.items(library.id):
  File "/app/app/modules/emby/__init__.py", line 142, in mediaserver_items
    tmdbid=int(item.get("tmdbid")) if item.get("tmdbid") else None,
           ^^^^^^^^^^^^^^^^^^^^^^^
ValueError: invalid literal for int() with base 10: '507941-section-375-marzi-ya-jabardasti'
INFO:    transfer.py - 开始执行下载器文件转移 ...
DEBUG:   __init__.py - 请求模块执行：scheduler_job ...
DEBUG:   __init__.py - 请求模块执行：list_torrents ...
DEBUG:   connectionpool.py - Resetting dropped connection: 192.168.0.15
DEBUG:   connectionpool.py - http://192.168.0.15:8080 "POST /api/v2/torrents/info HTTP/1.1" 200 2
INFO:    transfer.py - 没有获取到已完成的下载任务

**Comment by vvnocode** [comment]
> @vvnocode 请规范tmdbid命名


和tmdbid有关系吗？我不是太懂。我是从搜索出来的themoviedb数据里面做的资源检索。按道理来说，程序已经把下载和tmdbid关联了的吧。

**Comment by yubanmeiqin9048** [comment]
@vvnocode tmdbid应为纯数字，507941-section-375-marzi-ya-jabardasti => 507941

**Comment by cqboyone** [comment]
> @vvnocode tmdbid应为纯数字，507941-section-375-marzi-ya-jabardasti => 507941

我需要怎么修改呢

**Comment by yubanmeiqin9048** [comment]
假设你的文件夹是这样命名：`刑法典第375条 (2019) [tmdbid=507941-section-375-marzi-ya-jabardasti] `，修改tmdbid部分`刑法典第375条 (2019) [tmdbid=507941] `，并重新扫描该媒体所在的媒体库

**Comment by cqboyone** [comment]
> 假设你的文件夹是这样命名：`刑法典第375条 (2019) [tmdbid=507941-section-375-marzi-ya-jabardasti] `， 修改tmdbid部分`刑法典第375条 (2019) [tmdbid=507941] `

命名可以自动化吗？

**Comment by yubanmeiqin9048** [comment]
如果是想通过MP整理成规范形式，详见[readme](https://github.com/jxxghp/MoviePilot/blob/main/README.md) **进阶配置** 部分
如果是想解决该issue出现的问题你应该手动去规范现有资源命名或者重新通过MP整理

**Comment by cqboyone** [comment]
> 如果是想通过MP整理成规范形式，详见[readme](https://github.com/jxxghp/MoviePilot/blob/main/README.md) **进阶配置** 部分
> 
> 如果是想解决该issue出现的问题你应该手动去规范现有资源命名

为什么移动转移插件又可以完美识别下载转移无法识别的呢？

**Comment by vvnocode** [comment]
> > > mp映射目录问题给各位朋友参考： 威联通所有硬盘都挂在/share 目录下 使用qb套件版（默认下载器）、tr容器版（保种）。1、直接映射： /share：/share。 2、qb下载目录为： /share/downloads/, 下设 /电影 和 /电视剧 2个目录； 3、媒体库（link硬链接同步），目录为/share/downloads/media，下设 /电影和/电视剧目录。所有功能正常使用，字幕插件正常。
> > 
> > 
> > 下载路径和媒体路径分开可以吗？比如像我这样，
> 
> 无关映射，这是qb调获取完成种子api，没获取到数据，还没走到路径映射的逻辑。qb完成的种子有打标签吗，如有标签，上传完整的log日志，有助于更好的排查

我开启了调试模式，感觉发现了问题所在。moviepilot我开启的移动模式。qb客户端配置的做种限制，做种一分钟就删除种子（不想做种，开的永v）。所以一般来说在下载完成后，qb会删除种子，如果moviepilot是轮询查询的话，可能因为qb删除而无法查询到任务，因此无法转移。
我的设置在nastool是可以完美运行的，估计moviepilot的逻辑有了变化。请问有没有办法让qb恢复做种限制，我确实不想做种。

## 配置了做种限制的moviepilot日志

```log
NFO: transfer.py - 开始执行下载器文件转移 ...
DEBUG: init.py - 请求模块执行：scheduler_job ...
DEBUG: init.py - 请求模块执行：list_torrents ...
DEBUG: connectionpool.py - Resetting dropped connection: 192.168.0.15
DEBUG: connectionpool.py - http://192.168.0.15:8080/ "POST /api/v2/torrents/info HTTP/1.1" 200 2
INFO: transfer.py - 没有获取到已完成的下载任务
```

## 没有配置做种限制的日志

> 看样子是移动模式下moviepilot自动删除种子了。

```log
DEBUG:   __init__.py - 请求模块执行：transfer_completed ...
INFO:    command.py - 处理事件：transfer.complete - [<function ChineseSubFinder.download at 0x15258f5eb060>, <function WebHook.send at 0x15258f4ba520>]
DEBUG:   connectionpool.py - http://192.168.0.15:8080 "GET /api/v2/app/webapiVersion HTTP/1.1" 200 5
DEBUG:   connectionpool.py - http://192.168.0.15:8080 "POST /api/v2/torrents/addTags HTTP/1.1" 200 0
DEBUG:   connectionpool.py - http://192.168.0.15:8080 "POST /api/v2/torrents/delete HTTP/1.1" 200 0
INFO:    __init__.py - 移动模式删除种子成功：f22cdeead99488f433960125835b925e686f3efa 
INFO:    transfer.py - 下载器文件转移执行完成
```
